lib/liquid/tags/inline_comment.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Liquid
+  class InlineComment < Tag
+    def render_to_output_buffer(_context, output)
```

Comment:
Documenting our out of GitHub resolution here for potential future reader:

Adding inline comments inside tags and drops might be something we want to do but it would be hard. We don't want to _prevent_ that kind of feature from happening by introducing a tag that would make it impossible.

If we add the restriction that inline comment tags must all start their lines with a #, then we could implement inline comments everywhere later. Since our new tag here would be a subset of inline comments:

```liquid
{%
  # We're printing out a responsive image
  echo product
    | image_url:
      width: 500 # inline comment about why

      # We know the image is below the fold
      loading: 'lazy'
    | image_tag
%}
```

The alternative would treat the above as a single comment. Which would be a backwards incompatible change. If we treat them as errors now, we know the backward compatibility won't be a problem.

Reply:
I've made the code changes.